### DISCUSSION TITLE
How to preserve S3 Hive Metastore shcemas across trino versions?

### DISCUSSION THREAD

**Discussion by riazarbi** [trinodb/trino#11817] [Q&A] [answered]
Hello.

We run several s3-backed catalogs using the hive plugin. This works very well. However, when we upgrade trino to a newer release (which happens quite often) we often have to rebuild these catalogs from scratch, because trino doesn't want to work with a catalog whose schemas were created by an earlier version. Here's an example schema from one of our catalogs - 

```
{
  "writerVersion" : "364",
  "ownerName" : "datascience",
  "ownerType" : "USER",
  "parameters" : { }
}
```

It's the `writerVersion` that's the problem. 

Is there a way to upgrade a hive catalog to a newer version? I thought about regexing my objects but that seems... risky.

**Comment by findepi** [comment, marked as answer]
File metastore is primarily for testing/hacking, not really supposed to be used for persistent production state.
Please use a metastore service, like Glue.

**Reply by riazarbi** [reply]
Thanks @findepi . I actually came across this today, which explains a bit more of the background. We'll try move over to thrift.

https://github.com/trinodb/trino/issues/5714
